tmt/steps/prepare/shell.py

```diff
@@ -81,13 +86,34 @@ def go(
 
             environment.update(
                 topology.push(
-                    dirpath=self.step.plan.worktree,
+                    dirpath=workdir,
                     guest=guest,
                     logger=logger,
                     filename_base=filename_base))
 
+
+        prepare_wrapper_filename = f'{PREPARE_WRAPPER_FILENAME}.{guest.name}'
```

Comment:
The name should follow the same pattern as `filename_base` above, i.e. the filename base, specific for the purpose, followed by phase and guest *safe* names: `f'{PREPARE_WRAPPER_FILENAME}-{self.safe_name}-{guest.safe_name}'`.

Please, follow the same pattern, and I'll submit a patch adding a helper so we don't have to worry about such low-level details in the future.

Reply:
Yep, we need a helper function for this, I just noticed internal execute plugin follows a different pattern :)

Reply:
FTR, it's https://github.com/teemtee/tmt/pull/2424. Don't bother waiting for it, unless you really want to - I scheduled it for 1.30 milestone.

Reply:
Pushed new commit with the fix. I also fixed the "finish" wrapper script name with the same pattern. Thank you @happz !